app/upgrades/v1/bank_send.go

```diff
@@ -0,0 +1,158 @@
+			total_allocate_coin.Amount,
+			assumed_coin.Amount))
+	}
+
+	// after get total supply
+	after_total_supply := bankkeeper.GetSupply(ctx, "uguu")
+	ctx.Logger().Info(fmt.Sprintf("bank send : total supply[%d]", after_total_supply.Amount))
+
+	return nil
+}
+
+func tokenAllocation(
+	ctx sdk.Context,
+	authkeeper authkeeper.AccountKeeper,
+	bankkeeper bankkeeper.Keeper,
+	index int,
+	value BankSendTarget,
+	fromAddr sdk.AccAddress) sdk.Coin {
+	not_exist_vesting_account := true
+
+	// check exits VestingAccount
+	toAddr, err := sdk.AccAddressFromBech32(value.ToAddress)
+	if err != nil {
+		panic(err)
+	}
+	accI := authkeeper.GetAccount(ctx, toAddr)
+	if accI == nil {
+		panic(fmt.Sprintf("error address not exist: [%s][%s]", strconv.Itoa(index), value.ToAddress))
+	}
+
+	cont_acc, ok := accI.(*authvesting.ContinuousVestingAccount)
+	add_coin := sdk.NewCoin(value.Denom, sdk.NewInt(value.Amount))
+	if ok {
+		ctx.Logger().Info(fmt.Sprintf("bank send[%s] : ContinuousVestingAccount is exits [%s]", strconv.Itoa(index), cont_acc.String()))
+		not_exist_vesting_account = false
+
+		// 	add coins
+		cont_acc.OriginalVesting.Add(add_coin)
+		// start time sets a more past date.
+		if cont_acc.GetStartTime() > value.VestingStarts {
+			cont_acc.StartTime = value.VestingStarts
+		}
+		// end time sets a more future date.
+		if cont_acc.GetEndTime() < value.VestingEnds {
+			cont_acc.EndTime = value.VestingEnds
+		}
+
+		if err := cont_acc.Validate(); err != nil {
+			panic(fmt.Errorf("failed to validate ContinuousVestingAccount: %w", err))
+		}
+
+		authkeeper.SetAccount(ctx, cont_acc)
+	}
+
+	delayed_acc, ok := accI.(*authvesting.DelayedVestingAccount)
+	if ok {
+		ctx.Logger().Info(fmt.Sprintf("bank send[%s] : DelayedVestingAccount is exits [%s]", strconv.Itoa(index), delayed_acc.String()))
+		not_exist_vesting_account = false
+
+		// 	add coins
+		delayed_acc.OriginalVesting.Add(add_coin)
+
+		// end time sets a more future date.
+		if delayed_acc.GetEndTime() < value.VestingEnds {
+			delayed_acc.EndTime = value.VestingEnds
+		}
+		if err := delayed_acc.Validate(); err != nil {
+			panic(fmt.Errorf("failed to validate DelayedVestingAccount: %w", err))
+		}
+
+		authkeeper.SetAccount(ctx, delayed_acc)
+	}
+
+	if not_exist_vesting_account {
+		ctx.Logger().Info(fmt.Sprintf("bank send[%s] : ContinuousVestingAccount / DelayedVestingAccount not exits", strconv.Itoa(index)))
+		// not exist
+		// 	create vesting account
+		cont_vesting_acc := authvesting.NewContinuousVestingAccount(
+			accI.(*types.BaseAccount),
+			sdk.NewCoins(add_coin),
+			value.VestingStarts,
+			value.VestingEnds)
+
+		if err := cont_vesting_acc.Validate(); err != nil {
+			panic(fmt.Errorf("failed to validate new ContinuousVestingAccount: %w", err))
+		}
+
+		authkeeper.SetAccount(ctx, cont_vesting_acc)
+
+		if err := bankkeeper.SendCoins(ctx, fromAddr, toAddr, sdk.NewCoins(add_coin)); err != nil {
+			panic(err)
+		}
```

Comment:
In this implementation, sending actual token only happens when the account is not vesting account.
This causes problem with the total supply contradiction.

Reply:
@taiki1frsy 
I am sorry, but if possible, I would appreciate it if you could provide a solution.
I took this approach because it was the only way I could vest in my vesting account and become visible after vesting.
Does this mean that a VESTING account does not need a BANK SEND?
If so, how will the vesting tokens be visible in that account after vesting?